components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -124,6 +127,20 @@ constexpr char kNotificationAdUrlPrefix[] = "https://www.brave.com/ads/?";
 
 const base::Feature kServing{"AdServing", base::FEATURE_ENABLED_BY_DEFAULT};
 
+std::vector<std::string> ExtraCommandLineSwitches() {
+  std::vector<std::string> extra_command_line_switches;
+
+  const auto* const command_line = base::CommandLine::ForCurrentProcess();
+  if (command_line && command_line->HasSwitch(brave_rewards::kSwitchName)) {
+    const std::string command_line_switch = base::StrCat(
+        {brave_rewards::kSwitchName, "=",
+         command_line->GetSwitchValueASCII(brave_rewards::kSwitchName)});
```

Comment:
If you really want to copy over the command line arguments, then I would suggest putting some of this logic into `RewardsFlags`, e.g.

```cpp
struct RewardsFlags {
  // ..
  static std::string ExtractCommandLineSwitch(const base::CommandLine& command_line);
  // ..
};
```

Reply:
Fixed